src/lib/sql/gds_cache.hpp

```diff
@@ -87,6 +87,19 @@ class GDSCache : public AbstractCache<Key, Value> {
     this->_capacity = capacity;
   }
 
+  void resize(size_t capacity) {
+    if (_queue.size() > capacity) {
+      for (size_t difference = 0; difference < _queue.size() - capacity; ++difference) {
+        auto top = _queue.top();
+
+        _inflation = top.priority;
+        _map.erase(top.key);
+        _queue.pop();
```

Comment:
Seeing this code again, shouldn't each cache implement a protected `_evict` method? That way, `resize` could be part of the `AbstractCache`.

Reply:
You mean the Abstract cache would have a loop like `while (_queue.size > capacity)` calling `_evict()` inside? I see two problems. The loop condition depends on the cache type. Furthermore, there is cache dependent resize code, e.g., the random cache's reinitialization of the random generator.

Reply:
Ok, I missed that part about the implementation being different. I would still put the eviction part in its own method so that it can be shared with `set`

Reply:
Done